R/tokenization.R

```diff
@@ -13,4 +13,478 @@
 # limitations under the License.
 
```

Comment:
Please consider breaking this apart into separate files for readability. I can imagine wanting to have the "main" tokenizer functions in one window and the utilities in another.

Reply:
yup!